packages/genesys-spark-components/src/components/beta/gux-calendar-beta/components/single/gux-calendar.tsx

```diff
@@ -86,6 +86,17 @@ export class GuxCalendar {
   }
 
   private onDateClick(date: Date): void {
+    const currentDateElement: HTMLTableCellElement =
+      this.root.shadowRoot.querySelector(
+        `.gux-content-date[data-date="${date.getTime()}"]`
+      );
+    const isCurrentDateDisabled = Array.from(
+      currentDateElement.classList
+    ).includes('gux-disabled');
+    if (isCurrentDateDisabled) {
+      return;
+    }
+
```

Comment:
This seems like a very round about way to check this. We have a `date: Date` parameter and `minValue: Date` and `maxValue: Date` properties so we should be able to check it if is a valid date or not directly.

In the `getMonthDays` function we check this to get the disabled property already.
```
 const disabled =
        (this.minValue && currentDate.getTime() <= this.minValue.getTime()) ||
        (this.maxValue && currentDate.getTime() > this.maxValue.getTime());
```
We can move this into a function for reuse. I generally prefer "positive" functions like `isValidDate` not `isInvalidDate` but either will obviously work.

Reply:
Ah good call. Thanks for pointing this out.